tests/kani/AsyncAwait/spawn.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//
+// compile-flags: --edition 2018
+
+//! This file tests the executor and spawn infrastructure from the Kani library.
+
+use std::sync::{
+    atomic::{AtomicI64, Ordering},
+    Arc,
+};
+
+#[kani::proof]
+#[kani::unwind(4)]
+fn deterministic_schedule() {
+    let x = Arc::new(AtomicI64::new(0)); // Surprisingly, Arc verified faster than Rc
+    let x2 = x.clone();
+    kani::spawnable_block_on(
+        async move {
+            let x3 = x2.clone();
+            kani::spawn(async move {
+                x3.fetch_add(1, Ordering::Relaxed);
```

Comment:
If I understood this correctly, this should always runs first. In which case, you could assert that x3 is 0, right?

Reply:
Well, it depends how exactly the round robin is implemented, i.e. in what order tasks are added to the task list. I'm not sure we want to make any guarantees other than that it's deterministic

Reply:
As per discussion above, added the assert.